File: RealLiteUi.TypeScript/html/pagina/PagMissaoConclusao.ts
/// <reference path="../../../RealLifeDominio.TypeScript/MissaoConclusaoDominio.ts"/>

module RealLifeUi
{
    // #region Importações

    import MissaoConclusaoDominio = RealLifeDominio.MissaoConclusaoDominio;

    // #endregion Importações

    // #region Enumerados
    // #endregion Enumerados

    export class PagMissaoConclusao extends PagRealLifeBase
    {
        // #region Constantes
        // #endregion Constantes

        // #region Atributos

        private static _i: PagMissaoConclusao;

        public static get i(): PagMissaoConclusao
        {
            if (PagMissaoConclusao._i != null)
            {
                return PagMissaoConclusao._i;
            }

            PagMissaoConclusao._i = new PagMissaoConclusao();

            return PagMissaoConclusao._i;
        }

        private _divMissaoDinheiro: DivMissaoDinheiro;
        private _divMissaoResultado: DivMissaoResultado;
        private _divMissaoRp: DivMissaoRp;
        private _divMissaoRpProgresso: DivMissaoRpProgresso;
        private _objMissaoConclusao: MissaoConclusaoDominio;

        private get divMissaoDinheiro(): DivMissaoDinheiro
        {
            if (this._divMissaoDinheiro != null)
            {
                return this._divMissaoDinheiro;
            }

            this._divMissaoDinheiro = new DivMissaoDinheiro();

            return this._divMissaoDinheiro;
        }

        private get divMissaoResultado(): DivMissaoResultado
        {
            if (this._divMissaoResultado != null)
            {
                return this._divMissaoResultado;
            }

            this._divMissaoResultado = new DivMissaoResultado();

            return this._divMissaoResultado;
        }

        private get divMissaoRp(): DivMissaoRp
        {
            if (this._divMissaoRp != null)
            {
                return this._divMissaoRp;
            }

            this._divMissaoRp = new DivMissaoRp();

            return this._divMissaoRp;
        }

        private get divMissaoRpProgresso(): DivMissaoRpProgresso
        {
            if (this._divMissaoRpProgresso != null)
            {
                return this._divMissaoRpProgresso;
            }

            this._divMissaoRpProgresso = new DivMissaoRpProgresso();

            return this._divMissaoRpProgresso;
        }

        public get objMissaoConclusao(): MissaoConclusaoDominio
        {
            return this._objMissaoConclusao;
        }

        public set objMissaoConclusao(objMissaoConclusao: MissaoConclusaoDominio)
        {
            this._objMissaoConclusao = objMissaoConclusao;
        }

        // #endregion Atributos

        // #region Construtores
        // #endregion Construtores

        // #region Métodos

        public animar(): void
        {
            if (this.objMissaoConclusao == null)
            {
                return;
            }

            this.divMissaoResultado.animar();
        }

        public animar2(): void
        {
            this.divMissaoDinheiro.animar();
        }

        public animar3(): void
        {
            this.divMissaoRp.animar();
        }

        public animar4(): void
        {
            this.divMissaoRpProgresso.animar();
        }

        protected inicializar(): void
        {
            super.inicializar();

            this.divMissaoDinheiro.iniciar();
            this.divMissaoResultado.iniciar();
            this.divMissaoRp.iniciar();
            this.divMissaoRpProgresso.iniciar();

            this.testar();
        }

        private testar(): void
        {
            this.objMissaoConclusao = new MissaoConclusaoDominio();

            this.objMissaoConclusao.intDinheiro = 10000;
            this.objMissaoConclusao.intNivelAtual = 0;
            this.objMissaoConclusao.intReputacao = 250;

            this.animar();
        }

        // #endregion Métodos

        // #region Eventos
        // #endregion Eventos
    }
}

File: RealLifeUi/Html/Pagina/PagMissaoConclusao.cs
using NetZ.Web.Html;
using NetZ.Web.Server.Arquivo.Css;
using RealLifeUi.Html.Componente.Missao;

namespace RealLifeUi.Html.Pagina
{
    internal class PagMissaoConclusao : PagRealLifeBase
    {
        #region Constantes

        #endregion Constantes

        #region Atributos

        private DivMissaoDinheiro _divMissaoDinheiro;
        private DivMissaoResultado _divMissaoResultado;
        private DivMissaoRp _divMissaoRp;
        private DivMissaoRpProgresso _divMissaoRpProgresso;

        private DivMissaoDinheiro divMissaoDinheiro
        {
            get
            {
                if (_divMissaoDinheiro != null)
                {
                    return _divMissaoDinheiro;
                }

                _divMissaoDinheiro = new DivMissaoDinheiro();

                return _divMissaoDinheiro;
            }
        }

        private DivMissaoResultado divMissaoResultado
        {
            get
            {
                if (_divMissaoResultado != null)
                {
                    return _divMissaoResultado;
                }

                _divMissaoResultado = new DivMissaoResultado();

                return _divMissaoResultado;
            }
        }

        private DivMissaoRp divMissaoRp
        {
            get
            {
                if (_divMissaoRp != null)
                {
                    return _divMissaoRp;
                }

                _divMissaoRp = new DivMissaoRp();

                return _divMissaoRp;
            }
        }

        private DivMissaoRpProgresso divMissaoRpProgresso
        {
            get
            {
                if (_divMissaoRpProgresso != null)
                {
                    return _divMissaoRpProgresso;
                }

                _divMissaoRpProgresso = new DivMissaoRpProgresso();

                return _divMissaoRpProgresso;
            }
        }

        #endregion Atributos

        #region Construtores

        public PagMissaoConclusao() : base("Missão (conclusão)")
        {
        }

        #endregion Construtores

        #region Métodos

        protected override void montarLayout()
        {
            base.montarLayout();

            this.divMissaoResultado.setPai(this);
            this.divMissaoDinheiro.setPai(this);
            this.divMissaoRp.setPai(this);
            this.divMissaoRpProgresso.setPai(this);
        }

        protected override void setCss(CssArquivoBase css)
        {
            base.setCss(css);

            this.addCss(css.setBackgroundColor("black"));
            this.addCss(css.setColor("white"));
            this.addCss(css.setPaddingLeft(50));
            this.addCss(css.setPaddingRight(50));
            this.addCss(css.setPaddingTop(50));
        }

        #endregion Métodos

        #region Eventos

        #endregion Eventos
    }
}